src/test/java/org/folio/rest/unit/ItemDamagedStatusAPIUnitTest.java

```diff
@@ -0,0 +1,333 @@
+    when(pgClientFactory.getInstance(any(Context.class), any(Map.class))).thenReturn(postgresClient);
+  }
+
+  @Test
+  public void getItemDamagedStatusesShouldReturnBadRequest(TestContext testContext) {
+    RuntimeException exception = new RuntimeException(INTERNAL_SERVER_ERROR_MSG);
+    doAnswer(setExceptionForHandlerArgument(6, exception))
+      .when(postgresClient)
+      .get(
+        eq(REFERENCE_TABLE),
+        eq(ItemDamageStatus.class),
+        eq(new String[]{"*"}),
+        any(CQLWrapper.class),
+        eq(true),
+        eq(true),
+        any(Future.class)
+      );
+
+    Async async = testContext.async();
+    Future<Response> responseFuture = Future.future();
+    itemDamagedStatusAPI.getItemDamagedStatuses(
+      DEFAULT_QUERY,
+      DEFAULT_OFFSET,
+      DEFAULT_LIMIT,
+      DEFAULT_LANGUAGE,
+      DEFAULT_HEADERS,
+      responseFuture.completer(),
+      rule.vertx().getOrCreateContext()
+    );
+
+    responseFuture.setHandler(it -> {
+      testContext.assertEquals(400, it.result().getStatus());
+      async.complete();
+    });
+  }
+
+  @Test
+  public void getItemDamagedStatusesShouldReturnInternalServerError(TestContext testContext) {
+
+    doThrow(new RuntimeException(INTERNAL_SERVER_ERROR_MSG))
+      .when(postgresClient).get(
+      eq(REFERENCE_TABLE),
+      eq(ItemDamageStatus.class),
+      eq(new String[]{"*"}),
+      any(CQLWrapper.class),
+      eq(true),
+      eq(true),
+      any(Future.class)
+    );
+
+    Async async = testContext.async();
+    Future<Response> responseFuture = Future.future();
+    itemDamagedStatusAPI.getItemDamagedStatuses(
+      DEFAULT_QUERY,
+      DEFAULT_OFFSET,
+      DEFAULT_LIMIT,
+      DEFAULT_LANGUAGE,
+      DEFAULT_HEADERS,
+      responseFuture.completer(),
+      rule.vertx().getOrCreateContext()
+    );
+    responseFuture.setHandler(it -> {
+      testContext.assertEquals(500, it.result().getStatus());
+      async.complete();
+    });
+  }
+
+  @Test
+  public void postItemDamagedStatusesShouldReturnBadRequest(TestContext testContext) {
+    GenericDatabaseException exception = new GenericDatabaseException(new ErrorMessage(getUUIDErrorMap()));
+    doAnswer(setExceptionForHandlerArgument(3, exception))
+      .when(postgresClient)
+      .save(
+        eq(REFERENCE_TABLE),
+        anyString(),
+        any(ItemDamageStatus.class),
+        any(Future.class)
+      );
+
+    Async async = testContext.async();
+    Future<Response> responseFuture = Future.future();
+    itemDamagedStatusAPI.postItemDamagedStatuses(
+      DEFAULT_LANGUAGE,
+      new ItemDamageStatus(),
+      DEFAULT_HEADERS,
+      responseFuture.completer(),
+      rule.vertx().getOrCreateContext()
+    );
+
+    responseFuture.setHandler(it -> {
+      testContext.assertEquals(400, it.result().getStatus());
+      async.complete();
+    });
+  }
+
+  @Test
+  public void postItemDamagedStatusesShouldReturnInternalServerError(TestContext testContext) {
```

Comment:
The test name does not describe why an internal server error should be returned. Under what circumstances should it return an internal server error, when a specific exception is thrown?

Reply:
renamed